misc/ls-extensions/modules/sa-modeling-service/src/main/java/io/ballerina/componentmodel/servicemodel/nodevisitors/StatementNodeVisitor.java

```diff
@@ -0,0 +1,150 @@
+        return serviceId;
+    }
+
+    public String getConnectorType() {
+        return connectorType;
+    }
+
+    @Override
+    public void visit(VariableDeclarationNode variableDeclarationNode) {
+        if (variableDeclarationNode.typedBindingPattern().bindingPattern().toString().trim().equals(clientName)) {
+            TypeDescriptorNode typeDescriptorNode = variableDeclarationNode.typedBindingPattern().typeDescriptor();
+            connectorType = getClientModuleName(typeDescriptorNode);
+            NodeList<AnnotationNode> annotations = variableDeclarationNode.annotations();
+            this.serviceId = Utils.getId(annotations);
+        }
+    }
+
+    @Override
+    public void visit(ObjectFieldNode objectFieldNode) {
+        if (objectFieldNode.fieldName().text().trim().equals(clientName)) {
+            TypeDescriptorNode typeDescriptorNode = (TypeDescriptorNode) objectFieldNode.typeName();
+            connectorType = getClientModuleName(typeDescriptorNode);
+            Optional<MetadataNode> metadataNode = objectFieldNode.metadata();
+            if (metadataNode.isPresent()) {
+                NodeList<AnnotationNode> annotationNodes = metadataNode.get().annotations();
+                serviceId = Utils.getId(annotationNodes);
+            }
+        }
+    }
+
+    @Override
+    public void visit(ModuleVariableDeclarationNode moduleVariableDeclarationNode) {
+        if (moduleVariableDeclarationNode.typedBindingPattern().bindingPattern().toString().trim().equals(clientName)) {
+            TypeDescriptorNode typeDescriptorNode =
+                    moduleVariableDeclarationNode.typedBindingPattern().typeDescriptor();
+            connectorType = getClientModuleName(typeDescriptorNode);
+            Optional<MetadataNode> metadataNode = moduleVariableDeclarationNode.metadata();
+            if (metadataNode.isPresent()) {
+                NodeList<AnnotationNode> annotationNodes = metadataNode.get().annotations();
+                serviceId = Utils.getId(annotationNodes);
+            }
+        }
+    }
+
+    private String getClientModuleName(TypeDescriptorNode typeDescriptorNode) {
+        String clientModuleName = null;
+        if (typeDescriptorNode instanceof QualifiedNameReferenceNode) {
+            QualifiedNameReferenceNode clientNode = (QualifiedNameReferenceNode) typeDescriptorNode;
+            Optional<Symbol> listenerSymbol = semanticModel.symbol(clientNode);
+            if (listenerSymbol.isPresent() && (listenerSymbol.get() instanceof TypeReferenceTypeSymbol)) {
+                clientModuleName = ((TypeReferenceTypeSymbol)
+                        listenerSymbol.get()).signature().trim().replace(CLIENT, "");
+            } else {
+                clientModuleName = clientNode.modulePrefix().text().trim();
+            }
+        }
+        return clientModuleName;
+    }
+
+
+
+    @Override
+    public void visit(TypeDefinitionNode typeDefinitionNode) {
+
+    }
```

Comment:
```suggestion
    public void visit(TypeDefinitionNode typeDefinitionNode) {
        // no-op overloaded method to avoid unnecessary syntax tree visits
    }
```

Reply:
Resolved. Thank you